core/src/main/resources/lib/form/checkbox.jelly

```diff
@@ -74,18 +74,20 @@ THE SOFTWARE.
   <j:set var="name" value="${attrs.name ?: '_.'+attrs.field}"/>
   <j:set var="default" value="${attrs.default ?: false}"/>
   <j:set var="value" value="${attrs.checked ?: instance[attrs.field] ?: default}"/>
-
+  <l:js src="jsbundles/web-components.js" type="module"/>
   <span class="jenkins-checkbox">
-    <input type="checkbox"
-           name="${name}"
-           value="${attrs.value}"
-           title="${attrs.tooltip}"
-           onclick="${attrs.readonly=='true' ? 'return false;' : attrs.onclick}" id="${attrs.id}" class="${attrs.class} ${attrs.negative!=null ? 'negative' : null} ${attrs.checkUrl!=null?'validated':''}"
-           checkUrl="${attrs.checkUrl}" checkDependsOn="${attrs.checkDependsOn}" json="${attrs.json}"
-           disabled="${readOnlyMode ? 'true' : null}"
-           checked="${value ? 'true' : null}"/>
-    <label class="attach-previous ${attrs.title == null ? 'js-checkbox-label-empty' : ''}"
-           title="${attrs.tooltip}">${attrs.title}</label>
+    <jenkins-checkbox
+      tooltip="${attrs.tooltip}"
+      title="${attrs.title}"
+      name="${name}"
+      value="${attrs.value}"
+      disabled="${readOnlyMode ? 'true' : null}"
+      checked="${value ? 'true' : null}"
+      id="${attrs.id}"
+      onclick="${attrs.readonly=='true' ? 'return false;' : attrs.onclick}"
+      class="${attrs.class} ${attrs.negative!=null ? 'negative' : null} ${attrs.checkUrl!=null?'validated':''}"
+      checkUrl="${attrs.checkUrl}" checkDependsOn="${attrs.checkDependsOn}" json="${attrs.json}"
+    ></jenkins-checkbox>
```

Comment:
```suggestion
    />
```
conventionally

Reply:
web components must have closed fully, no self closing.

Reply:
:thinking: how can it matter? This is a Jelly XML file.

Reply:
i don't know enough about jelly. Does jelly automatically add the closing tag? or does it output it as is to the browser? if the second, then it needs to be closed in jelly too.

https://stackoverflow.com/questions/23961178/do-custom-elements-require-a-close-tag

Essentially only legacy tags are self closing (img, input, etc)

Reply:
> does it output it as is to the browser?

No, it parses XML, then generates (X)HTML. The input syntax should be irrelevant.

Reply:
The problem is the web component specification as pointed out by  @halkeye which does not allow self-closing, even if they do not have children.

Reply:
But again this source file is not a literal template; it is an XML document which is parsed. You can see this in the following passing test in a trivial plugin:

```java
package demo;
import hudson.Extension;
import hudson.model.InvisibleAction;
import hudson.model.RootAction;
@Extension public class X extends InvisibleAction implements RootAction {
    @Override public String getUrlName() {
        return "xxx";
    }
}
```

`X/index.jelly`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?jelly escape-by-default='true'?>
<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout">
      <l:layout>
          <l:main-panel>
              <xxx/>
          </l:main-panel>
      </l:layout>
</j:jelly>
```

```java
package demo;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.containsString;
import org.junit.Rule;
import org.junit.Test;
import org.jvnet.hudson.test.JenkinsRule;
public class XTest {
    @Rule public JenkinsRule r = new JenkinsRule();
    @Test public void tag() throws Exception {
        assertThat(r.createWebClient().goTo("xxx/").getWebResponse().getContentAsString(), containsString("<xxx></xxx>"));
    }
}
```